packages/core/src/redux/slices/notifications.ts

```diff
@@ -0,0 +1,90 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+import { EventEmitter } from "events";
+import { RootState } from "../store";
+import { createAppThunk } from "../thunk";
+
+export enum NotificationLogLevels {
+    "debug",
+    "log",
+    "info",
+    "warn",
+    "error",
+}
+
+export interface Notification {
+    type: string;
+    message: string;
+    level?: NotificationLogLevels;
+    props?: {
+        [key: string]: unknown;
+    };
+}
+
+export interface NotificationEvent extends Notification {
+    level: NotificationLogLevels;
+    timestamp: number;
+}
+
+export type NotificationsEventEmitter = EventEmitter<{ [key: string]: [NotificationEvent] }>;
```

Comment:
Could this be `{ [key in NotificationLogLevels]: ...`?

Reply:
We removed `NotificationLogLevels` in favour of using a [string literal union type](https://www.oreilly.com/library/view/learn-react-with/9781789610253/61996a69-dc93-4705-822e-1bb2fd01c873.xhtml). Therefore this comment is outdated.

Reply:
Yeah, so this can still work like this:
`[key in "debug" | "log" | "info" | "warn" | "error"]`

Reply:
And of course the literal type can be extracted to 
`type Level = "debug" | "log" | "info" | "warn" | "error";`

Reply:
Great suggestions and now added to PR ⭐